go-apps/meep-loc-serv/server/loc-serv.go

```diff
@@ -363,17 +378,21 @@ func checkNotificationRegisteredZoneStatus(zoneId string, apId string, nbUsersIn
 			}
 		}
 	}
+	mutex.Unlock()
 }
 
 func checkNotificationRegisteredUsers(oldZoneId string, newZoneId string, oldApId string, newApId string, userId string) {
 
+	mutex.Lock()
+
```

Comment:
Use this 
mutex.Lock()
defer mutex.Unlock()

defer will call Unlock before exiting the function... preventing having multiple mutex.Unlock() spread in the function.

Reply:
defer added everywhere to prevent future error.